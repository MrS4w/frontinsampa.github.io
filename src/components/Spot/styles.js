export default ({ palette }) => ({
  spot: {
    borderColor: palette.background.default,
    borderStyle: 'solid',
    borderWidth: 1,
    display: 'block',
    height: '100%',
    overflow: 'hidden',
    position: 'relative',
    width: '100%',

    '& > img': {
      height: 'inherit',
      left: '50%',
      position: 'relative',
      transform: 'translateX(-50%)',
      width: 'inherit',
    },
  },
});
