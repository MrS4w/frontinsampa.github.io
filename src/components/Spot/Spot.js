import React from 'react';
import PropTypes from 'prop-types';

import { withStyles } from '../../styles';

import styles from './styles';

const Spot = ({ classes, children }) => (
  <div className={classes.spot}>
    { children }
  </div>
);

Spot.propTypes = {
  classes: PropTypes.object.isRequired,
  children: PropTypes.node.isRequired,
};

export default withStyles(styles)(Spot);
