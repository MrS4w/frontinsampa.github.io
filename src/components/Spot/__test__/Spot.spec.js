import React from 'react';
import { shallow } from 'enzyme';

import Spot from '../Spot';

describe('Components: <Spot />', () => {
  const wrapper = shallow(
    <Spot classes={{}}>
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </Spot>,
  );

  it('Create a snapshot.', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
