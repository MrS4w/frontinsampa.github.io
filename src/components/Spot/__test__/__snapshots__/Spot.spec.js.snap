// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: <Spot /> Create a snapshot. 1`] = `
<Spot
  classes={
    Object {
      "spot": "Spot-spot-1 Spot-spot-2",
    }
  }
>
  Lorem ipsum dolor sit amet consectetur adipisicing elit.
</Spot>
`;
